src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -600,3 +680,37 @@ prefixes =
   , "$c"
   , "$m"
   ]
+
+
+safeTyThingForRecord :: TyThing -> Maybe (T.Text, [(T.Text, T.Text)])
+safeTyThingForRecord (AnId _) = Nothing
+safeTyThingForRecord (AConLike dc) =
+    let flds = conLikeFieldLabels dc
+        name = occName . conLikeName $ dc
+        types = map (conLikeFieldType dc . flLabel) flds
+        ctxStr = T.pack . showGhc $ name
+        fld_pairs = [(T.pack . unpackFS . flLabel $ x, T.pack . showGhc $ y) | x <- flds | y <- types]
+    in
+        Just (ctxStr, fld_pairs)
+        --[(showGhc name, [("x", "y")])]
+safeTyThingForRecord _ = Nothing
+
+mkRecordSnippetCompItem :: T.Text -> [(T.Text, T.Text)] -> T.Text -> SpanDoc -> CompItem
+mkRecordSnippetCompItem ctxStr compl mn docs = r
```

Comment:
Here `compl` contains a tuple (name, type) for every field, but It looks like the type is not used anywhere, so why bother to pass it through (and why bother collecting it)?

Reply:
I left it here since I was not sure if it would be nice to attach `type` information in the code snippet.

This is from your suggestion:

> That is, given a record `data Person = Person { name :: String}`:
> 
>     * your action currently suggests `Person{name = name}`,
> 
>     * instead, it would be better to suggest `Person{name = _name}`.

Should we consider, 
```
Person {name = (_name:String)}  
```
With this the user has information about the type as well. I do understand that the `type` is also displayed as part of `codeLens` documentation. So, in that sense, it could be redundant.

For now, I will go with your suggestion, unless you feel adding `type` would be an improvement.

Reply:
My feeling is that the type is redundant, and including it will not be trivial for polymorphic types. Moreover, the type might include identifiers that are not in scope, creating a build error.

Reply:
Sounds good. I will go with your suggestion. I think it is an improvement over what I have currently.

Reply:
Any ideas on how to handle field names which already have a prefixed `_`.

For example, `Person { _name:: String }`.  Should we leave it as is or add another `__name`?

Reply:
Not a big deal